pe_exceptions/utils/pe_exception_safety.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef PE_EXCEPTION_SAFETY_HPP
+#define PE_EXCEPTION_SAFETY_HPP
+
+#include <concepts>
+#include <ostream>
+#include <type_traits>
+
+namespace peach::detail
+{
+
+  // prevents possibility of a this pointer being passed in the format_error context
+  class pe_fatal_error;
+  inline std::ostream& operator<<( std::ostream& stream, [[maybe_unused]] const pe_fatal_error& ec )
+  {
+    return stream;
+  }
+
+  template< typename T > concept OutputStreamInsertable = requires( std::ostream& os, const T& arg )
+  {
+    {
+      os << arg
+    }
+    ->std::convertible_to< std::ostream& >;
```

Comment:
```suggestion
    { os << arg } -> std::convertible_to< std::ostream& >;
```

But you don't actually use this.

Reply:
I do in `pe_exception_format.hpp`, 
```cpp
  template< OutputStreamInsertable... Tys >
  std::string format_error( const std::size_t line_num, const std::string& file_name,
                            Tys&&... args )
```